test/parallel/test-watch-mode-files_watcher.mjs

```diff
@@ -51,6 +52,33 @@ describe('watch mode file watcher', () => {
     assert.strictEqual(changesCount, 1);
   });
 
+  it('should watch changed files with same prefix path string', async () => {
+    mkdirSync(tmpdir.resolve('subdir'));
+    mkdirSync(tmpdir.resolve('sub'));
+    const file1 = tmpdir.resolve('subdir', 'file1.mjs');
+    const file2 = tmpdir.resolve('sub', 'file2.mjs');
+    writeFileSync(file2, 'export const hello = () => { return "hello world"; };');
+    writeFileSync(file1, 'import { hello } from "../sub/file2.mjs"; console.log(hello());');
+
+    const child = spawn(process.execPath,
+                        ['--watch', file1],
+                        { stdio: ['ignore', 'pipe', 'ignore'] });
+    let completeCount = 0;
+    for await (const line of createInterface(child.stdout)) {
+      if (line.startsWith('Completed running')) {
+        completeCount++;
+      }
+      if (completeCount === 1) {
+        appendFileSync(file1, '\n // append 1');
+      }
+      // The file is reloaded due to file watching
+      if (completeCount === 2) {
+        child.kill();
+        await once(child, 'exit');
```

Comment:
nit: I missed that earlier, but we don't really need to wait for the `'exit'` event here, since we already await for `child.stdout` to close in the `for await`. We can also keep it, it's fine

```suggestion
```